### PR TITLE
Fix Google/Microsoft Sheets action-node config leaking across sibling nodes

### PR DESCRIPTION
**Pull request by bannarisoftwares**
## Summary

Follow-up to #223. While testing #223's fix live, found that a condition's "Yes" and "No" branch action nodes (both Google Sheets, or both Microsoft Excel) could end up pointing at the *same* spreadsheet/workbook, and that typed-but-unsaved fields (Plugin Name, Worksheet Name) reverted after clicking "Connect" — plus two issues surfaced by automated review on #223 itself.

- **Config leaking across sibling nodes** — `GoogleSheetsConfigForm` read `spreadsheetId`/`spreadsheetUrl` from local component state captured once on mount and never re-synced. `NodeConfigPanel` reuses the same form instance across different action nodes of the same plugin type (no remount when switching selection), so whichever node's spreadsheet was loaded first would get silently persisted onto whatever node was saved next. Fixed by reading those fields directly from `initialData` (always current) instead of stale local state — matching how Microsoft's form already worked.
- **Plugin Name / Worksheet Name reverting after Connect** — these fields are plain local React state until "Save action" is clicked, and the OAuth redirect is a full-page reload that discards them. Added `lib/pendingConfigFields.ts`, which stashes them to `sessionStorage` immediately before navigating and restores them when the form remounts after returning from Google/Microsoft.
- **File-naming parity** — Google Sheets' auto-created files had no distinguishing suffix (Microsoft's did). Added one, and fixed Microsoft's own suffix, which was accidentally derived from the run ID (changes every run) instead of the stable per-node ID — defeating its own purpose of telling sibling nodes' files apart.
- **Automated review feedback from #223** (both addressed here since they touch the same files):
  - The OAuth "just connected" latch (added in #223 to protect a freshly-parsed token from being clobbered by a sync effect) was held for the component's entire lifetime. Since switching to a different node doesn't remount the form, this could leak a token/name from one node onto another. Now reset via a new optional `instanceKey` prop (`NodeConfigPanel` passes `node.id`) whenever the selected node actually changes.
  - The two automation config writes (`Automation.graph` + `AutomationRun.graphSnapshot`) went through `Promise.all`, not atomic together — if one write failed after the other succeeded, they'd permanently diverge. Now both go through a single Prisma `$transaction`.
  - Corrected the `jsonSetNodeConfig` docstring, which overstated protection: the atomic UPDATE prevents concurrent writes to *different* nodes from clobbering each other, but not two concurrent *runs* of the *same* node racing on a stale read during first-ever auto-create (documented as an accepted narrow edge case — no response data is lost, only a possible orphaned spreadsheet).

## Test plan

- [x] `pnpm --filter backend exec tsc --noEmit` and `pnpm --filter form-app exec tsc --noEmit` — clean
- [x] `pnpm test:unit` — 2843/2843 backend tests pass, including new/updated coverage for the transaction wrap
- [x] `pnpm --filter backend lint` / `pnpm --filter form-app lint` — 0 issues in every file touched
- [x] Verified live: renamed a Google Sheets action node, disconnected/reconnected its Google account, confirmed the typed name survived the OAuth redirect and the node's spreadsheet ID stayed independent from its sibling branch node — checked directly against the database (`automation.graph` JSON), not just the UI
- [x] Pre-push hook ran full test suites + builds for backend, form-viewer, admin-app, and form-app — all green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Google Sheets and Microsoft Excel files now use distinguishing workbook/spreadsheet names to reduce ambiguity across multiple automation actions.
  - Plugin config forms persist unsaved “Plugin Name” and related fields across full-page OAuth redirects.

- **Bug Fixes**
  - Automation action node switching now better isolates configuration so values don’t leak between node instances.
  - Automation configuration persistence is now atomic, updating both the live graph and run snapshot together to avoid inconsistent saves.

- **Tests**
  - Added stronger assertions ensuring the automation graph updates occur within a single transaction.

### CODE REVIEW

File: apps/backend/src/services/automation/engine.ts [hunk 1]
```diff
@@ -125,25 +125,40 @@ function mergeStepOutput(context: unknown, nodeId: string, output: any): Automat
...
   nodeId: string,
   config: PluginConfig
-): Promise<void> {
+) {
   const tableIdent = Prisma.raw(`"${table}"`);
   const columnIdent = Prisma.raw(`"${column}"`);
   const configJson = JSON.stringify(config);
 
-  await prisma.$executeRaw(Prisma.sql`
+  return prisma.$executeRaw(Prisma.sql`
```
[the remark this answers is not in the document]
Comment (bannarisoftwares) [reply apps/backend/src/services/automation/engine.ts:161]: Fair point on layering, but deferring this one: every other Prisma call in this file (`prisma.automationRun.update`, `prisma.automationStepRun.create`, etc.) already accesses Prisma directly from this service — there's no existing automation repository layer to slot into, and introducing one here would refactor the whole file's established pattern rather than this PR's specific fix. Scoping a full repository extraction as a separate follow-up if the team wants to pursue it.
